Libraries/LibWeb/IndexedDB/Internal/Index.cpp

```diff
@@ -107,4 +108,18 @@ u64 Index::count_records_in_range(GC::Ref<IDBKeyRange> range)
     return count;
 }
 
+void Index::store_a_record(IndexRecord const& record)
+{
+    m_records.append(record);
+
+    // NOTE: The record is stored in index’s list of records such that the list is sorted primarily on the records keys, and secondarily on the records values, in ascending order.
```

Comment:
Sorting in inserted order is meant to be a stable sort, which quick sort is not. I'm not sure if this is observable for this case, but if so we need to use a stable sorting algorithm.

Reply:
The methods on `Index` and `ObjectStore` should be proxied to the database when that eventually gets implemented, so I think its fine to leave this as is for now.

Reply:
Even once we do that, will still need to access it in a way that meets that sorted criteria though, right? That would be the case whether it's done on insertion like this does, or if we're accessing the stored records based on that sorted criteria.

Or am I misunderstanding things? All pretty moot without tests behind it anyhow 😅

Reply:
Since inserting and sorting is a implementation detail of whatever database system we choose, that should happen for free afaik. 🤔 🤷 

However, i do not belive that a stable sort here matters.
The sort on the key would not be stable, but since we then sort on the values (and the values must be unique), it should always result in the same order.